jenkins/dockerized-e2e-runner.sh

```diff
@@ -60,8 +60,16 @@ if [[ "${KUBE_RUN_FROM_OUTPUT:-}" =~ ^[yY]$ ]]; then
     )
 fi
 
+# Timeouts can leak the container, causing weird issues where the job keeps
+# running and *deletes resources* that the next job is using.
+# Give the container a unique name, and make bash kill it on EXIT,
+# which will happen during a timeout.
+CONTAINER_NAME="${JOB_NAME}-${BUILD_NUMBER}"
+trap "docker kill ${CONTAINER_NAME}" EXIT
```

Comment:
there's `docker stop` that sends SIGTERM, then SIGKILL after 10 seconds-- does e2e-runner.sh do any useful recovery?